packages/client/src/AccountsClient.js

```diff
@@ -172,10 +172,14 @@ export class AccountsClient {
     }
 
     try {
+      const { onUserCreated } = this.options;
       const userId = await this.transport.createUser(user);
       if (callback && isFunction(callback)) {
         callback();
       }
+      if (isFunction(onUserCreated)) {
+        await onUserCreated({ id: userId });
```

Comment:
In case of an error, you want to prevent the creation of the user? or it's just a notification hook? (if so, it should be wrapped with try/catch, so it won't fail the user creation)